File: example/src/main/java/eu/darken/mvpbakery/example/AppComponent.java
package eu.darken.mvpbakery.example;


import android.app.Activity;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import eu.darken.mvpbakery.injection.ComponentSource;

@Singleton
@Component(modules = {
        AppModule.class,
        ActivityBinderModule.class,
        AndroidInjectionModule.class
})
public interface AppComponent {

    ExampleApplication inject(ExampleApplication application);

    ComponentSource<Activity> activityComponentSource();

    @Component.Builder
    interface Builder {
        Builder appModule(AppModule module);

        AppComponent build();
    }

}

File: example/src/main/java/eu/darken/mvpbakery/example/AppComponent.kt
package eu.darken.mvpbakery.example


import android.app.Activity
import dagger.Component
import dagger.android.AndroidInjectionModule
import eu.darken.mvpbakery.injection.ComponentSource
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, ActivityBinderModule::class, AndroidInjectionModule::class])
interface AppComponent {

    fun inject(application: ExampleApplication): ExampleApplication

    fun activityComponentSource(): ComponentSource<Activity>

    @Component.Builder
    interface Builder {
        fun appModule(module: AppModule): Builder

        fun build(): AppComponent
    }

}
